ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala

```diff
@@ -826,29 +828,47 @@ private class PostgresLedgerDao(
           Option[WorkflowId],
           Date,
           InputStream,
-          Option[InputStream]))(implicit conn: Connection) =
+          Option[InputStream],
+          InputStream))(implicit conn: Connection) =
     contractResult match {
-      case (coid, transactionId, workflowId, ledgerEffectiveTime, contractStream, keyStreamO) =>
+      case (coid, transactionId, workflowId, ledgerEffectiveTime, contractStream, keyStreamO, tx) =>
         val witnesses = lookupWitnesses(coid)
         val divulgences = lookupDivulgences(coid)
+        val absoluteCoid = AbsoluteContractId(coid)
+
+        val (signatories, observers) =
+          transactionSerializer
+            .deserializeTransaction(ByteStreams.toByteArray(tx))
```

Comment:
This makes every contract lookup when validating exercises much more expensive. Not sure what to do about it though.

Reply:
I believe the only possible solution would be to add two columns holding the stakeholders. Would this be feasible? If so, what would the right procedure be to add new "metadata" columns?

Reply:
One easy place to add it to is `ContractInst` (which already gets serialized), but it might not be right place actually. For example, `NodeCreate` holds the stakeholders at the same level as the contract value of type `ContractInst`.

@S11001001, @remyhaemmerle-da: what do you think?

Reply:
Created issue to discuss about this: for the moment, I believe we can merge this. https://github.com/digital-asset/daml/issues/2330